Mathlib/Topology/UniformSpace/Dini.lean

```diff
@@ -0,0 +1,141 @@
+/-
+Copyright (c) 2024 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import Mathlib.Analysis.Normed.Order.Lattice
+import Mathlib.Topology.ContinuousMap.Ordered
+import Mathlib.Topology.UniformSpace.CompactConvergence
+
+/-! # Dini's Theorem
+
+This file proves Dini's theorem, which states that if `F n` is a monotone increasing sequence of
+continuous real-valued functions on a compact set `s` converging pointwise to a continuous function
+`f`, then `F n` converges uniformly to `f`.
+
+We generalize the codomain from `ℝ` to a normed latttice additive commutative group `G`.
+This theorem is true in a different generality as well: when `G` is a linearly ordered topological
+group with the order topology. This weakens the assumptions of the norm and commutativity, in
+exchange for strengthening to a linear order. This separate generality is not included in this file,
+but that generality was included in initial drafts of the original
+[PR #19068](https://github.com/leanprover-community/mathlib4/pull/19068) and can be recovered if
+necessary.
+
+-/
+
+open Filter Topology
+
+variable {ι α G : Type*} [Preorder ι] [IsDirected ι (· ≤ ·)] [Nonempty ι]
```

Comment:
Turns out you can remove *both* directedness and nonemptyness by casing on triviality of `atTop` like this:
```
rcases (atTop : Filter ι).eq_or_neBot with h | _
  · simp only [h, tendstoLocallyUniformly_iff_forall_tendsto, bot_prod, tendsto_bot, implies_true]
```
(This would be even shorter if we had `tendstoLocallyUniformly_bot`)